### ISSUE TITLE
Latest release doesn't appear in Fdroid repo

### ISSUE THREAD

**Issue by kumy** [cgeo/cgeo-fdroid#8]
We expect `2019.05.20` to be available, but it's not. However the job ran fine http://ci.cgeo.org/view/FDroid/job/Publish%20FDroid%20-%20RELEASE/46/console

Raised in https://github.com/cgeo/cgeo/issues/7534#issuecomment-494307928
cc @Lomanic

**Comment by Lineflyer** [comment]
The update is available for me on fdroid. Also another user in the mentioned issue confirmed availability.

However I do see two warnings, where I am not sure why it is there and if it was there before:

https://f-droid.org/wiki/page/Antifeature:KnownVuln

https://f-droid.org/wiki/page/Antifeature:DisabledAlgorithm

Is there something wrong with our app signing?

**Comment by kumy** [comment]
Strange, the file `cgeo-release-2019.05.20.apk` was on filesystem since May 20 20:38 CEST and indexes from the same date :/.

If I'm not mistaken, there was quite the same behavior last time with NB, that was not available for everyone. Is this a bug on our side or in F-Droid not correctly refreshing indexes?

**Comment by kumy** [comment]
I personally don't have the update available and I can see my logs in the webserver access logs:

```
92.xxxxx - - [21/May/2019:18:04:43 +0000] "HEAD /nightly/index-v1.jar HTTP/1.1" 200 0 "-" "F-Droid 1.6" 1250542 "Host-fdroid-cgeo-org-56" "http://xxxx:80" 0ms
92.xxxxx - - [21/May/2019:18:04:43 +0000] "HEAD /index-v1.jar HTTP/1.1" 200 0 "-" "F-Droid 1.6" 1250544 "Host-fdroid-cgeo-org-56" "http://xxxx:80" 0ms
```

I really suspect a bug in FDroid

**Comment by kumy** [comment]
It started working for me after I disabled both repo in Fdroid (`mainline` and `nightly`) then reenabled `mainline`.

**Comment by kumy** [comment]
@Lineflyer regarding the warning, both seems to be related to the apk itself. Apparently one of our dependencies or sdk or "something" is still using a vulnerable OpenSSL version it should be updated. But I didn't found exactly the source of the issue. There is plenty of threads on the net. BTW didn't google play complains about it?

The other seems to be a problem when we sign the apk itself, but one more time, I don't know exactly what to check. `apksigner` return this:
```
# apksigner verify -v --print-certs repo/mainline/repo/cgeo-release-2019.05.20.apk
Verifies
Verified using v1 scheme (JAR signing): true
Verified using v2 scheme (APK Signature Scheme v2): true
Number of signers: 1
Signer #1 certificate DN: CN=mucek4, OU=gorenc, O=org, L=Gorenjska, ST=Si, C=SI
Signer #1 certificate SHA-256 digest: 3db1dd8443b96fe9f7680eea4f7f9d0234f8407283ffe3e6a43ad68adcfdc5e4
Signer #1 certificate SHA-1 digest: 449be14e69f3c98cac748ae09118a2454701839c
Signer #1 certificate MD5 digest: a98b7c446a267acd7e0a67ff3f451dcc
Signer #1 key algorithm: RSA
Signer #1 key size (bits): 2048
Signer #1 public key SHA-256 digest: 9ccc42332bf2fa4a8dc8f4158150e25a7f275c4dba46ea0ba122f375a766312e
Signer #1 public key SHA-1 digest: e7d83153f0fb73af2869312dc89e40232ad89f57
Signer #1 public key MD5 digest: f25a2c225abbfa910e6eed00acd9ef11
WARNING: META-INF/CHANGES not protected by signature. Unauthorized modifications to this JAR entry will not be detected. Delete or move the entry outside of META-INF/.
WARNING: META-INF/README.md not protected by signature. Unauthorized modifications to this JAR entry will not be detected. Delete or move the entry outside of META-INF/.
WARNING: META-INF/services/com.fasterxml.jackson.core.JsonFactory not protected by signature. Unauthorized modifications to this JAR entry will not be detected. Delete or move the entry outside of META-INF/.
WARNING: META-INF/services/com.fasterxml.jackson.core.ObjectCodec not protected by signature. Unauthorized modifications to this JAR entry will not be detected. Delete or move the entry outside of META-INF/.
```

Does the warning means lines `Signer #1 certificate MD5 digest: a98b7c446a267acd7e0a67ff3f451dcc` should be absent? Do we have to update our singing keys?

And finaly, the F-droid index are signed [using SHA256](http://ci.cgeo.org/view/FDroid/job/Publish%20FDroid%20-%20RELEASE/46/console).

Those discussion should be moved to main repo, as it doesn't seems to be related to fdroid.

**Comment by kumy** [comment]
I tend to close this as disabling enabling sources fixes the problem. @Lineflyer?

**Comment by Lineflyer** [comment]
Yup..lets close for now. For me it was working instantly after the publish with a short normal refresh (pull down to refresh in f-droid).

**Comment by Lomanic** [comment]
Deleting and re-adding the repository, I was eventually able to update cgeo via f-droid. Probably some caching or bug on the client side indeed, not on the repository. Thanks for investigating @kumy and sorry for the bogus report.

**Comment by githubber42** [comment]
For a "long time" f-droid was not showing the nightly builds for me, so I gave up on it and just clicked the message on cgeo home screen to download new apks. For the last 2 days, f-droid IS working for me now. A few things have changed - newer LineageOS without full microg, but with UnifiedNLP (Dejavu and Nominatim-OSM, new f-droid client 1.7-alpha0, so it's not clear what brings the improvement, but here is my hypothesis. Maybe someone could test it. Or maybe the new client fixed a bug. :D

Installing cgeo the first time from a browser-downloaded apk sets something so f-droid does not properly update. In the past, I did not always install cgeo first from f-droid. Sometimes yes, sometimes no; sometimes f-droid updates worked, sometimes it did not work. The check/uncheck repository in f-droid was not a reliable "fix" for me. This time I made sure my first install of c:geo was from f-droid, and it is now working for 2 days in a row.

Other interesting bits, to me: 1) Without microg, cgeo "alert" messages DO display. I assumed they needed GSF, and would fail. 2) With orbot set to VPN mode and cgeo "checked" to use tor, logins to GC DO work. Using desktop or phone torbrowser, GC logins fail for me.
